src/Features/Core/Portable/CodeFixes/CodeFixService.cs

```diff
@@ -266,17 +271,14 @@ internal partial class CodeFixService : ICodeFixService
 
             if (fixes != null && fixes.Any())
             {
+                // If the fix provider supports fix all occurrences, then get the corresponding FixAllProviderInfo and fix all context.
+                var fixAllProviderInfo = extensionManager.PerformFunction(fixer, () => ImmutableInterlocked.GetOrAdd(ref _fixAllProviderMap, fixer, FixAllProviderInfo.Create), defaultValue: null);
+
                 FixAllCodeActionContext fixAllContext = null;
-                var codeFixProvider = fixer as CodeFixProvider;
-                if (codeFixProvider != null)
+                if (fixAllProviderInfo != null)
                 {
-                    // If the codeFixProvider supports fix all occurrences, then get the corresponding FixAllProviderInfo and fix all context.
-                    var fixAllProviderInfo = extensionManager.PerformFunction(codeFixProvider, () => ImmutableInterlocked.GetOrAdd(ref _fixAllProviderMap, codeFixProvider, FixAllProviderInfo.Create), defaultValue: null);
-
-                    if (fixAllProviderInfo != null)
-                    {
-                        fixAllContext = FixAllCodeActionContext.Create(document, fixAllProviderInfo, codeFixProvider, diagnostics, this.GetDocumentDiagnosticsAsync, this.GetProjectDiagnosticsAsync, cancellationToken);
-                    }
+                    var codeFixProvider = (fixer as CodeFixProvider) ?? new WrapperCodeFixProvider((ISuppressionFixProvider)fixer, diagnostics);
```

Comment:
Can the caller create this wrapper?  if yes, I guess we can keep the above dictionary key as CodeFixProvider and we won't need to do this casting here?